GitUI/Avatars/InitialsAvatarGenerator.cs

```diff
@@ -32,33 +32,59 @@ public Image GenerateAvatarImage(string email, string name, int imageSize)
 
         protected internal (string initials, int hashCode) GetInitialsAndHashCode(string email, string name)
         {
-            string initials;
-            int hashCode;
+            string initials = null;
+            int hashCode = _unkownCounter;
             if (!string.IsNullOrWhiteSpace(name))
             {
-                var trimmedName = name.Trim();
-                var indexSpace = trimmedName.IndexOf(' ');
-                initials = trimmedName[0].ToString().ToUpper();
-                if (indexSpace != -1)
+                initials = GetInitials(name.Trim().Split());
+                if (!string.IsNullOrWhiteSpace(initials))
                 {
-                    initials += trimmedName[indexSpace + 1].ToString().ToUpper();
+                    hashCode = name.GetHashCode();
+                    return (initials, hashCode);
                 }
-
-                hashCode = string.IsNullOrWhiteSpace(email) ? name.GetHashCode() : email.GetHashCode();
-            }
-            else if (!string.IsNullOrWhiteSpace(email))
-            {
-                initials = ("" + email.TrimStart().First()).ToUpper();
-                hashCode = email.GetHashCode();
             }
-            else
+
+            if (!string.IsNullOrWhiteSpace(email))
             {
-                initials = "?";
-                hashCode = _unkownCounter;
-                _unkownCounter++;
+                var e = email.TrimStart();
+                if (e.Contains("@"))
+                {
+                    e = e.Split('@')[0];
+                }
+
+                initials = GetInitials(e.Split('.'));
```

Comment:
Should we split only on '.'? or should we also support '_' and '-' (and probably some others)?

Reply:
I have seen uses like abc_sd.fgfhhf@ and thought that was inconsistent. But it is maybe better to use first, last also here?
Will change to .-_

Reply:
updated. This is a fallback case, most users have some kind of  name set. The heuristics to select the initials from the name couls possibly be improved.